airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/job_factory/OAuthConfigSupplier.java

```diff
@@ -0,0 +1,93 @@
+ * SOFTWARE.
+ */
+
+package io.airbyte.scheduler.persistence.job_factory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Strings;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.DestinationOAuthParameter;
+import io.airbyte.config.SourceOAuthParameter;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.UUID;
+
+public class OAuthConfigSupplier {
+
+  final private ConfigRepository configRepository;
+
+  public OAuthConfigSupplier(ConfigRepository configRepository) {
+    this.configRepository = configRepository;
+  }
+
+  public JsonNode injectSourceOAuthParameters(UUID sourceDefinitionId, UUID workspaceId, JsonNode sourceConnectorConfig)
+      throws IOException {
+    try {
+      configRepository.listSourceOAuthParam().stream()
+          .filter(p -> sourceDefinitionId.equals(p.getSourceDefinitionId()))
+          .filter(p -> p.getWorkspaceId() == null || workspaceId.equals(p.getWorkspaceId()))
+          // we prefer params specific to a workspace before global ones (ie workspace is null)
+          .min(Comparator.comparing(SourceOAuthParameter::getWorkspaceId, Comparator.nullsLast(Comparator.naturalOrder()))
+              .thenComparing(SourceOAuthParameter::getOauthParameterId))
+          .ifPresent(
+              sourceOAuthParameter -> injectJsonNode((ObjectNode) sourceConnectorConfig, (ObjectNode) sourceOAuthParameter.getConfiguration()));
+      return sourceConnectorConfig;
+    } catch (JsonValidationException e) {
+      throw new IOException(e);
+    }
+  }
+
+  public JsonNode injectDestinationOAuthParameters(UUID destinationDefinitionId, UUID workspaceId, JsonNode destinationConnectorConfig)
+      throws IOException {
+    try {
+      configRepository.listDestinationOAuthParam().stream()
+          .filter(p -> destinationDefinitionId.equals(p.getDestinationDefinitionId()))
+          .filter(p -> p.getWorkspaceId() == null || workspaceId.equals(p.getWorkspaceId()))
+          // we prefer params specific to a workspace before global ones (ie workspace is null)
+          .min(Comparator.comparing(DestinationOAuthParameter::getWorkspaceId, Comparator.nullsLast(Comparator.naturalOrder()))
+              .thenComparing(DestinationOAuthParameter::getOauthParameterId))
+          .ifPresent(destinationOAuthParameter -> injectJsonNode((ObjectNode) destinationConnectorConfig,
+              (ObjectNode) destinationOAuthParameter.getConfiguration()));
+      return destinationConnectorConfig;
+    } catch (JsonValidationException e) {
+      throw new IOException(e);
+    }
+  }
+
+  private static void injectJsonNode(ObjectNode config, ObjectNode fromConfig) {
+    for (String key : Jsons.keys(fromConfig)) {
+      if (!config.has(key) || isSecretMask(config.get(key).asText())) {
```

Comment:
When are we inserting that the secrets mask is necessary to be overwritten?

Reply:
I'm proposing to persist in the connector's configuration with injected and masked OAuth parameters in this PR: https://github.com/airbytehq/airbyte/pull/5865

This would be done by web backend endpoints of the API when creating a connector.

The motivation is to have a full connector configuration saved by the UI, so JSON/spec validation can verify that the connector's configuration is sound.